### PR TITLE
Create Installer enhancement for GCP private/internal clusters.

### PR DESCRIPTION
**Pull request by patrickdillon**
This enhancement extends OpenShift Installer to support installating Internal Private only clusters on GCP which are only accessible from VPC network and not visible to the Internet.

/assign @sdodson
/cc @abhinavdahiya 
/cc @kalexand-rh

### CODE REVIEW

File: enhancements/installer/gcp-private-clusters.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+Many customer environments don't require external connectivity from the outside world and as such, they would prefer not to expose the cluster endpoints to public. Currently the OpenShift installer exposes endpoints of the cluster like Kubernetes API, or OpenShift Ingress to the Internet, although most of these endpoints can be made Internal after installation with varying degree of ease individually, creating OpenShift clusters which are Internal by default is highly desirable for users.
+
+Specifically for GCP, this enhancement requires implementation of internal load balancers, which were not an initial feature on the GCP platform.
+
+## Motivation
+
+### Goals
+Install an OpenShift cluster on GCP as internal/private so that it is only accessible from within an internal network and not visible to the public Internet.
+
```
Comment (abhinavdahiya) [inline enhancements/installer/gcp-private-clusters.md]: nit: extra lines here

File: enhancements/installer/gcp-private-clusters.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+## Summary
+
+Many customer environments don't require external connectivity from the outside world and as such, they would prefer not to expose the cluster endpoints to public. Currently the OpenShift installer exposes endpoints of the cluster like Kubernetes API, or OpenShift Ingress to the Internet, although most of these endpoints can be made Internal after installation with varying degree of ease individually, creating OpenShift clusters which are Internal by default is highly desirable for users.
+
+Specifically for GCP, this enhancement requires implementation of internal load balancers, which were not an initial feature on the GCP platform.
+
+## Motivation
+
+### Goals
+Install an OpenShift cluster on GCP as internal/private so that it is only accessible from within an internal network and not visible to the public Internet.
```
Comment (abhinavdahiya) [inline enhancements/installer/gcp-private-clusters.md]: nit: leave a new line

File: enhancements/installer/gcp-private-clusters.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+1. **Merge at each milestone.** Merge when the design is able to transition to a
+   new status (provisional, implementable, implemented, etc.). View anything
+   marked as `provisional` as an idea worth exploring in the future, but not
+   accepted as ready to execute. Anything marked as `implementable` should
+   clearly communicate how an enhancement is coded up and delivered. If an
+   enhancement describes a new deployment topology or platform, include a
+   logical description for the deployment, and how it handles the unique aspects
+   of the platform. Aim for single topic PRs to keep discussions focused. If you
+   disagree with what is already in a document, open a new PR with suggested
+   changes.
```
Comment (patrickdillon) [inline enhancements/installer/gcp-private-clusters.md]: oops. will delete above section.

File: enhancements/installer/gcp-private-clusters.md (hunk 1 above)
Comment (patrickdillon) [reply enhancements/installer/gcp-private-clusters.md]: removed

File: enhancements/installer/gcp-private-clusters.md (hunk 2 above)
Comment (patrickdillon) [reply enhancements/installer/gcp-private-clusters.md]: added new line after `### Goals`

**Comment by patrickdillon** [comment]
I will go through and add a new line after each header.

**Comment by patrickdillon** [comment]
Whitespace should be in order, so let me know if there are any issues now.

File: enhancements/installer/gcp-private-clusters.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+pullSecret: '{"auths": ...}'
+```
+
+##### Basedomain
+
+`Basedomain` will continue to be used for creating a private DNS zone and the necessary records, but the requirement for a public DNS zone for the basedomain will be removed.
+
+##### Public Subnets
+
+No public subnets need to be provided to the installer.
```
Comment (abhinavdahiya) [inline enhancements/installer/gcp-private-clusters.md]: on GCP there is no public vs private subnet, only control-plane and compute ??

Comment (patrickdillon) [reply enhancements/installer/gcp-private-clusters.md]: I removed this section.

File: enhancements/installer/gcp-private-clusters.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+##### Internal Load Balancers and Instance Groups
+
+The GCP platform was originally setup to use only network load balancers (NLBs) in order to ensure necessary health checks, but it is not possible to limit access to external load balancers based on source tags[(see GCP firewall docs)][gcp-firewall-sources]. Therefore, it is necessary to implement internal load balancers to allow access to internal instances. 
+
+The Internal Load Balancer relies on Instance Groups rather than the Target Pools used by the NLB. The installer will create instance groups for each zone, even if there is no instance in that group. This difference will need to be taken into account by the [cluster-api-provider-gcp][gcp-provider]. 
+
+###### Other Resources to Be Created Differently
+
+* Cluster IP - The cluster IP address will be internal only. 
+* Forwarding rule - One forwarding rule will handle both the API & IGN ports.
```
Comment (abhinavdahiya) [inline enhancements/installer/gcp-private-clusters.md]: nit `API` -> `Kubernetes API`
`IGN` -> `machine config server`

File: enhancements/installer/gcp-private-clusters.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+---
+
+# installing.internal.private.clusters.for.gcp
+
+## Release Signoff Checklist
+
+- [ X ] Enhancement is `implementable`
+- [ X ] Design details are appropriately documented from clear requirements
+- [ X ] Test plan is defined
+- [ X ] Graduation criteria for dev preview, tech preview, GA
```
Comment (abhinavdahiya) [inline enhancements/installer/gcp-private-clusters.md]: I think `[x]` renders correctly.

File: enhancements/installer/gcp-private-clusters.md (hunk 5 above)
Comment (patrickdillon) [reply enhancements/installer/gcp-private-clusters.md]: done

File: enhancements/installer/gcp-private-clusters.md (hunk 6 above)
Comment (patrickdillon) [reply enhancements/installer/gcp-private-clusters.md]: done

**Pull request merged**